sympy/core/function.py

```diff
@@ -279,6 +280,10 @@ def __repr__(cls):
         return cls.__name__
 
 
+class ExprFunctionClass(FunctionClass, ExprManagedProperties):
+    pass
```

Comment:
There is `BooleanFunction` which is defined as
```python
class BooleanFunction(Application, Boolean):
    pass
```
This means that `Application` cannot be an `Expr` subclass and so its metaclass `FunctionClass` must subclass `ManagedProperties`. Then `Function` is defined as
```python
class Function(Application, Expr, metaclass=ExprFunctionClass):
    pass
```
where we need this separate metaclass to distinguish from that of `BooleanFunction`.